lib/WP_Auth0_Options.php

```diff
@@ -129,92 +134,124 @@ public function get_logout_url() {
 		return add_query_arg( 'action', 'logout', site_url( 'wp-login.php', 'login' ) );
 	}
 
+	/**
+	 * Get as a string-separated string and parse to array
+	 *
+	 * @return array
+	 */
+	public function get_lock_connections() {
+		$connections = $this->get( 'lock_connections' );
+		$connections = empty( $connections ) ? array() : explode( ',', $connections );
+		return array_map( 'trim', $connections );
+	}
+
+	/**
+	 * Add a new connection to the lock_connections setting
+	 *
+	 * @param string $add_conn - connection name to add
+	 */
+	public function add_lock_connection( $add_conn ) {
+		$connections = $this->get_lock_connections();
+
+		// Add if it doesn't exist already
+		if ( ! array_key_exists( $add_conn, $connections ) ) {
```

Comment:
This is a suggestion:
It could be more performant if you FIRST check that the string from `$this->get( 'lock_connections' );` "contains" the trimmed `$add_conn` parameter.
- If it DOES contain it, return (skip). This way, you avoid converting it to an array in the first place.
- if not, then you can call the `get_lock_connections()` method that prepares the array for you.

Reply:
You can't really do that can you? What if I want to add the connection `google-oauth` so I search for the string `google` in `twitter,facebook,google-oauth2` and it's there so I bail out?